crates/starknet/src/state/mod.rs

```diff
@@ -72,80 +164,89 @@ impl StateChanger for StarknetState {
         address: ContractAddress,
         class_hash: ClassHash,
     ) -> DevnetResult<()> {
-        let addr: Address = address.into();
-        let persistent_state = Arc::make_mut(&mut self.state.state_reader);
+        let persistent_state = &mut self.state.state;
 
-        persistent_state.address_to_class_hash_mut().insert(addr.clone(), class_hash.bytes());
-        persistent_state.address_to_nonce_mut().insert(addr, Felt252::new(0));
+        persistent_state.address_to_class_hash.insert(address, class_hash);
+        persistent_state.address_to_nonce.insert(address, Felt::from(0));
 
         Ok(())
     }
 
     fn change_storage(&mut self, storage_key: ContractStorageKey, data: Felt) -> DevnetResult<()> {
-        let persistent_state = Arc::make_mut(&mut self.state.state_reader);
+        let persistent_state = &mut self.state.state;
 
-        persistent_state.address_to_storage_mut().insert(storage_key.into(), data.into());
+        persistent_state.address_to_storage.insert(storage_key, data);
 
         Ok(())
     }
 
     fn increment_nonce(&mut self, address: ContractAddress) -> DevnetResult<()> {
-        let addr: Address = address.into();
-        let nonce = self.state.get_nonce_at(&addr)?;
-        let persistent_state = Arc::make_mut(&mut self.state.state_reader);
+        let nonce = self.state.state.get_nonce_at(address.try_into()?)?;
```

Comment:
Why do we first get the nonce here? To induce some kind of error if address doesn't exist? If so, could that be commented?

Reply:
fixed in https://github.com/0xSpaceShard/starknet-devnet-rs/commit/e1ae43cd3ca34f8d478f6a310e0e84c921d38158